test/system/src/test/java/io/pravega/test/system/MultiControllerTest.java

```diff
@@ -98,24 +108,34 @@ public void getControllerInfo() {
         log.info("Controller Service direct URI: {}", controllerURIDirect);
         controllerURIDiscover.set(URI.create("pravega://" + String.join(",", uris)));
         log.info("Controller Service discovery URI: {}", controllerURIDiscover);
-        URI controllerURI = URI.create("tcp://" + String.join(",", uris));
-        log.info("Controller Service direct URI: {}", controllerURI);
-        controllerClient = new ControllerImpl(ControllerImplConfig.builder()
-                                                                  .clientConfig(ClientConfig.builder()
-                                                                                            .controllerURI(controllerURI)
-                                                                                            .build())
-                                                                  .build(),
-                                              EXECUTOR_SERVICE);
+        controllerClientDirect = new ControllerImpl(ControllerImplConfig.builder()
+                .clientConfig(ClientConfig.builder()
+                        .controllerURI(controllerURIDirect.get())
+                        .build())
+                .build(),
+                EXECUTOR_SERVICE);
+        controllerClientDiscover = new ControllerImpl(ControllerImplConfig.builder()
+                .clientConfig(ClientConfig.builder()
+                        .controllerURI(controllerURIDiscover.get())
+                        .build())
+                .build(),
+                EXECUTOR_SERVICE);
+
+        segmentStoreService = Utils.createPravegaSegmentStoreService(zkUris.get(0), controllerURIDirect.get());
+        log.debug("Pravega host service details: {}", segmentStoreService.getServiceDetails());
+
     }
 
     @After
-    public void tearDown() {
+    public void tearDown() throws ExecutionException {
         ExecutorServiceHelpers.shutdown(EXECUTOR_SERVICE);
         if (controllerService1 != null && controllerService1.isRunning()) {
```

Comment:
If the test explicitly kills all the instances of the controller in `multiControllerTest`, are the lines inside the `if` block going to be executed ever? This impacts on whether clients are closed or not.

Reply:
using atomic reference for controller client, and removed the earlier line from the if block.